code/go/0chain.net/blobbercore/filestore/tree_validation.go

```diff
@@ -0,0 +1,436 @@
+// This file will validate challange tree and validation tree and provide merkle path for the provided
+// index/es as well.
+
+package filestore
+
+import (
+	"encoding/hex"
+	"errors"
+	"io"
+	"math"
+
+	"github.com/0chain/gosdk/core/util"
+	"golang.org/x/crypto/sha3"
+)
+
+const (
+	KB = 1024
+	MB = KB * KB
+)
+
+const (
+	HashSize = 32
+	// FMTSize is the size of tree nodes of fixed merkle tree excluding root node
+	FMTSize     = 65472
+	FMTDisKRead = 10 * MB
+)
+
+// fixedMerkleTree is used to verify fixed merkle tree root and store tree nodes to the file
+type fixedMerkleTree struct {
+	*util.FixedMerkleTree
+}
+
+// getNodesSize will calculate the size required to store tree nodes excluding root node.
+func getNodesSize(dataSize, merkleLeafSize int64) int64 {
+	totalLeaves := (dataSize + merkleLeafSize - 1) / merkleLeafSize
+	if totalLeaves == 1 {
+		return 0
+	}
+	totalNodes := totalLeaves
```

Comment:
We are calculating total number leaves and intermediate leaves excluding root leaf.
So if there are 10 leaves then total nodes would be 10 + 5 + 3 + 2 = 20.

Reply:
so you want to say that you will have different proof length for different number of leaves? I don't think it is how fixed merkle trees work, they should have the fixed length of proof, since the height is always the same

Reply:
@dabasov Actually this getNodesSize is not used by FixedMerkleTree. It is used by ValidationTree. However it will work for both trees.
What `CalculateRootAndStoreNodes` does is, it calculates whole tree and store that tree in the file. `getNodesSize` will return total size required to store all the nodes of tree including leaves but excluding root leaf.

And for fixedMerkleTree like you said, it will be fixed length. So I've used FMTSize constant.